x-pack/legacy/plugins/reporting/export_types/common/lib/screenshots/types.ts

```diff
@@ -37,14 +36,21 @@ export interface Screenshot {
   description: any;
 }
 
-export interface ScreenShotOpts {
```

Comment:
I'm confused on this deletion: did these go anywhere or are they no longer needed?

Reply:
I removed the `Opts` references from these lines: https://github.com/elastic/kibana/pull/46851/files#diff-1a70afc14f0154fde139ff3e5eeca902L116-L133 because they didn't provide a value of safety, and they were creating hard-to-read syntax.

They got into the code originally to prevent `NOTE: Typescript does not throw an error if this interface has errors!
` which is a new comment in this PR. But Typescript **just does not** match the observable's payload data with what the function should return, so that code isn't saving us from mistakes.

As a replacement for all those `Opts` types, I kept the type references there just for the first time it is referenced. 

So in:
```
(browser, metrics: PerformanceMetrics | null) => ({ browser, metrics })
```
is the first time the `metrics` variable was referenced. In:
```
({ browser, metrics }, timeRange: TimeRange | null) => ({ browser, metrics, timeRange })
```
is the first time the `timeRange` variable is referenced

Typescript seems to be able to understand that the output of one mergeMap should match the input of the next mergeMap, but I haven't found a way to make it complain if the observable isn't what the function says it should return: `Rx.Observable<ScreenshotResults>`